packages/celotool/src/lib/geth.ts

```diff
@@ -667,6 +674,99 @@ export const simulateClient = async (
   }
 }
 
+const getBigData = (size: number) => {
+  return '0x' + '00'.repeat(size)
+}
+
+const getTxConf = async (dataTest: boolean) => {
+  if (dataTest) {
+    return {
+      feeCurrencyGold: true,
+      tokenName: 'cGLD.L',
+      transferFn: transferCalldata,
+    }
+  }
+
+  // randomly choose which token to use
+  const useGold = Boolean(Math.round(Math.random()))
+  const _transferFn = useGold ? transferCeloGold : transferCeloDollars
+  const _tokenName = useGold ? 'cGLD' : 'cUSD'
+
+  // randomly choose which gas currency to use
+  const _feeCurrencyGold = Boolean(Math.round(Math.random()))
+  return {
+    feeCurrencyGold: _feeCurrencyGold,
+    tokenName: _tokenName,
+    transferFn: _transferFn,
+  }
+}
+
+const getNonce = async (
+  kit: ContractKit,
+  senderAddress: string,
+  lastTx: any,
+  lastNonce: any,
+  gasPrice: BigNumber,
+  lastGasPriceMinimum: BigNumber
+) => {
+  let _nonce, _newPrice
+  _newPrice = gasPrice
+  if (lastTx === '' || lastNonce === -1) {
+    _nonce = await kit.web3.eth.getTransactionCount(senderAddress, 'latest')
+  } else if ((await kit.connection.getTransactionReceipt(lastTx))?.blockNumber) {
+    _nonce = await kit.web3.eth.getTransactionCount(senderAddress, 'latest')
+  } else {
+    _nonce = (await kit.web3.eth.getTransactionCount(senderAddress, 'latest')) - 1
+    _newPrice = BigNumber.max(gasPrice.toNumber(), lastGasPriceMinimum.times(1.15)).dp(0)
+    console.warn(
+      `TX ${lastTx} was not mined. Replacing tx reusing nonce ${_nonce} and gasPrice ${_newPrice}`
+    )
+  }
+  return {
+    newPrice: _newPrice,
+    nonce: _nonce,
+  }
+}
+
+const transferFnCatch = (error: any, baseLogMessage: any) => {
```

Comment:
I would rename this function.
It's the transferCatch yes, but it's not clear why it's returning true if it needs an account unlock.